pyxdsm/XDSM.py

```diff
@@ -325,9 +353,18 @@ def connect(
         if (not isinstance(label_width, int)) and (label_width is not None):
             raise ValueError("label_width argument must be an integer")
 
+        sys_faded = {}
+        for s in self.systems:
+            sys_faded[s.node_name] = s.faded
+        if (self.auto_fade["connections"] == "all") or (
+            self.auto_fade["connections"] == "connected"
+            and ((src in sys_faded and sys_faded[src]) or (target in sys_faded and sys_faded[target]))
+        ):
+            faded = True
+
```

Comment:
Can we alter the inputs/logic so that connections coming from a given component are treated differently from those going into it? When I have used fading in XDSM's I went with the convention of unfading a connection when the source component was unfaded, even if the target is still faded

Reply:
For example:

![Screenshot from 2022-10-03 11-32-14](https://user-images.githubusercontent.com/20371123/193617527-1fe2121a-1b7e-4215-af84-c3f8e0a3dfd6.png)

Reply:
Implemented, you can now provide `"incoming"` or `"outgoing"` as options for `"connections"`